CougSat1-PMIC/src/events/EventPeriodic.cpp

```diff
@@ -17,9 +17,629 @@
+	}
+	else if (BattCurrB > 4)
+	{
+		for(int i = 0; i < 12; i++)
+			{
+				// iterate through payload arrays
+				// use sizeof nodesPRBatt[] as a condition
+				// to make sure there is not out of bounds error
+				if(i < 7)
+				{
+					// get current and priority number
+					// of each payload
+					nodesPR3V3[i].getcurrent(&RegPR);
+					nodesPR3V3[i].getPriority(&priority1);
+
+					nodesPRBatt[i].getCurrent(&BattPR);
+					nodesPRBatt[i].getPriority(&priority2);
+					
+					// calculate x Value
+					X1 = (priority1 * RegPR);     
+					X2 = (priority2 * priority2); 
+
+					if(X1 < X2)
+						nodesPR3V3[i].setSwitch(true, false);
+					else
+						nodesPRBatt[i].setSwitch(true,false);
+									
+				}
+				else
+				{
+					nodesPR3V3[i].getcurrent(&RegPR);
+					nodesPR3V3[i].getPriority(&priority1);
+					X1 = (priority1 * RegPR); 
+					if( (i+1) != 12)
+					{		
+						nodesPR3V3[i + 1].getcurrent(&RegPR);
+						nodesPR3V3[i + 1].getPriority(&priority1);				
+						X2 = (priority1 * RegPR);// calculate x Vaule
+
+						// if X1 is greater than X2, save the index value
+						// else save index value of payload 2
+						if(X1 > X2)
+							x_index = i;
+						else
+							x_index = i+1;
+					}
+					nodesPV3V3[x_index].setSwitch(true,false);
+				}
+			}
+	}
+
+
+
+	
+	/*******************************************
+	 * Monitor Temperature of the Batteries    *
+	 * Use thermistorBattA/B and getTemperature*
+	 * to get temperature of both batteries    *
+	 * Turn on heater if temperature is below  *
+	 * TEMP_MIN. Turn of heaters when          *
+	 * temperature is above (TEMP_MIN + 10)	   *
+	 ****************************************/
+	result = thermistorBattA.getTemperature(&temp1);
+	if (result != ERROR_SUCCESS)
+	{
+		ERROR("Periodic", "Failed to get temperature of battery A");
+		return result;
+	}
+
+	result = thermistorBattB.getTemperature(&temp2);
+	if (result != ERROR_SUCCESS)
+	{
+		ERROR("Periodic", "Failed to get temperature of Battery B");
+		return result;
+	}
+
+	if (temp1 < TEMP_MIN)
+	{
+		// battery A is cold, turn on heater
+		if (BattVoltA < BattVoltB)
+			// power heater A with with battery B
+			nodesBatteryHeaters[0].setSwitch(false, true);
+		else if (BattVoltA > BattVoltB)
+			// Heater powered by battery A
+			nodesBatteryHeaters[0].setSwitch(true, false);
+		else
+			nodesBatteryHeaters[0].setSwitch(true, true);
+	}
+	else if (temp2 < TEMP_MIN)
+	{
+		// battery B is cold, turn on heater
+		if (BattVoltB < BattVoltA)
+			// power heater B with battery A
+			nodesBatteryHeaters[1].setSwitch(true, false);
+		else if (BattVoltB > BattVoltA)
+			// Heater powered by battery B
+			nodesBatteryHeater[1].setSwitch(false, true);
+		else
+			// both batteries power heater
+			nodeBatteryHeaters[1].setSwitch(true, true);
+	}
+	else
+	{
+		// if temperature reads 10 degrees higher than min temp or if temp reads higher
+		// higher than max temp, turn of heaters.
+		if( temp1 > (TEMP_MIN + 10) || temp1 > TEMP_MAX)
```

Comment:
I like the hysteresis, move the magic number to PMICConfiguration.
If a battery is too hot then its load or charging needs to be reduced